sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -1,209 +1,221 @@
-   * - `dataTransformer`: A set of `encode`/`decode` methods to be used to encode an event before sending to service
-   * and to decode the event received from the service
-   * - `userAgent`      : A string to append to the built in user agent string that is passed as a connection property
-   * to the service.
-   * - `websocket`      : The WebSocket constructor used to create an AMQP connection if you choose to make the connection
-   * over a WebSocket.
-   * - `webSocketConstructorOptions` : Options to pass to the Websocket constructor when you choose to make the connection
-   * over a WebSocket.
-   * - `retryOptions`   : The retry options for all the operations on the client/producer/consumer.
-   * A simple usage can be `{ "maxRetries": 4 }`.
+   * Subscribes to multiple partitions.
+   *
+   * Use this overload if you want to read from a subset of partitions and not coordinate with
+   * other subscribers.
+   *
+   * @param partitionIds The partitions IDs to subscribe to
    */
-  constructor(
-    host: string,
-    eventHubName: string,
-    consumerGroupName: string,
-    credential: TokenCredential,
-    options?: EventHubClientOptions
-  );
-  constructor(
-    param1: string,
-    param2: string,
-    param3?: string | EventHubClientOptions,
-    param4?: TokenCredential | EventHubClientOptions,
-    options?: EventHubClientOptions
-  ) {
-    this._consumersMap = new Map();
-    if (isTokenCredential(param4)) {
-      this._client = new EventHubClient(param1, param2, param4, options);
-      this._consumerGroupName = param3 as string;
-    } else if (typeof param3 === "string") {
-      this._client = new EventHubClient(param1, param2, param4);
-      this._consumerGroupName = param3;
+  subscribe(onReceivedEvents: OnReceivedEvents, options?: SubscriptionOptions): Subscription;
+  subscribe(partitionIds: string[], onReceivedEvents: OnReceivedEvents, options?: SubscriptionOptions): Subscription;
+  subscribe(
+    partitionManager: PartitionManager,
+    onReceivedEvents: OnReceivedEvents,
+    options?: SubscriptionOptions
+  ): Subscription;
+  subscribe(
+    onReceivedEventsOrPartitionIdsOrPartitionManager:
+      | OnReceivedEvents
+      | string[]
+      | PartitionManager,
+    optionsOrOnReceivedEvents: OnReceivedEvents | SubscriptionOptions | undefined,
+    possibleOptions?: SubscriptionOptions
+  ): Subscription {
+    let eventProcessor: EventProcessor;
+
+    if (typeof onReceivedEventsOrPartitionIdsOrPartitionManager === "function") {
+      // 1st constructor - read from all partitions, don't coordinate
+      const partitionProcessorType = createPartitionProcessorType(
+        onReceivedEventsOrPartitionIdsOrPartitionManager as OnReceivedEvents,
+        optionsOrOnReceivedEvents as SubscriptionOptions
+      );
+
+      eventProcessor = new EventProcessor(
+        this.consumerGroupName,
+        this._eventHubClient,
+        partitionProcessorType,
+        new InMemoryPartitionManager()
```

Comment:
Is the partition manager still needed if we're using the `GreedyPartitionLoadBalancer`? Or is it non-trivial to make it optional in EventProcessor?

Reply:
Probably not - I think this points a bit more to having a better split of responsibilities between the classes. 

I'm probably not going to be doing this in this PR, however. :)